Bindings/Python/examples/convert_muscle_type.py

```diff
@@ -0,0 +1,111 @@
+#                                                                         #
+# Licensed under the Apache License, Version 2.0 (the "License");         #
+# you may not use this file except in compliance with the License.        #
+# You may obtain a copy of the License at                                 #
+# http://www.apache.org/licenses/LICENSE-2.0.                             #
+#                                                                         #
+# Unless required by applicable law or agreed to in writing, software     #
+# distributed under the License is distributed on an "AS IS" BASIS,       #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         #
+# implied. See the License for the specific language governing            #
+# permissions and limitations under the License.                          #
+# ----------------------------------------------------------------------- #
+
+# convert_muscle_type.py
+# Author: Ayman Habib
+# ------------------------------------------------------------------------#
+# ABSTRACT: This short piece of OpenSim python API example demonstrates   #
+# how to load a model in the OpenSim python API, cycle through the        #
+# the muscles in the model and replace those of type Thelen2003 with      #
+# the more modern (true to publication) Millard2012AccelerationMuscle     #
+# There are differencs between the two muscle models so the resulting     #
+# model may need to be tuned manually.
+# ------------------------------------------------------------------------#
+
+import opensim as osim
+
+import sys
+# Helper function to copy Properties between two objects
+def copy_common_muscle_properties(from_muscle, to_muscle):
+    print ("Copying Properties for muscle:" +from_muscle.getName())
+    # Common properties are in base class Muscle this includes
+    # - max_isometric_force
+    # - optimal_fiber_length
+    # - tendon_slack_length
+    # - pennation_angle_at_optimal
+    # - max_contraction_velocity
+    # - ignore_tendon_compliance
+    # - ignore_activation_dynamics
+    to_muscle.copyProperty_max_isometric_force(from_muscle)
+    to_muscle.copyProperty_optimal_fiber_length(from_muscle)
+    to_muscle.copyProperty_tendon_slack_length(from_muscle)
+    to_muscle.copyProperty_pennation_angle_at_optimal(from_muscle)
+    to_muscle.copyProperty_max_contraction_velocity(from_muscle)
+    to_muscle.copyProperty_ignore_tendon_compliance(from_muscle)
+    to_muscle.copyProperty_ignore_activation_dynamics(from_muscle)
+    # Properties inherited from PathActuator
+    to_muscle.set_GeometryPath(from_muscle.get_GeometryPath().clone())
+    to_muscle.copyProperty_optimal_force(from_muscle)
+    # Properties from ScalarActuator
+    to_muscle.copyProperty_max_control(from_muscle)
+    to_muscle.copyProperty_min_control(from_muscle)
+    # Properties from Force
+    to_muscle.copyProperty_appliesForce(from_muscle)
+    # print (to_muscle.dump())
```

Comment:
I am not sure I understand the rationale for using the low-level property machinery to do the copying opposed to using the interface provided by the Components themselves. You clearly need to know the differences between Muscles anyway (cannot map directly from one to the other derived Muscle model) but you could easily dynamic cast both to a common base (e.g. Muscle) and use the assignment operator. Then use public getters and setters to handle the leftover which are only in the derived classes.